extension/data.ts

```diff
@@ -940,60 +940,57 @@ class RcxDict {
   }
 
   makeText(entry: DictEntryData | null, max: number): string {
-    let e;
-    let i;
-    let j;
-    let t;
-
     if (entry === null) {
       return '';
     }
 
-    const b = [];
+    const result = [];
 
     if (entry.kanji) {
-      b.push(entry.kanji + '\n');
-      b.push((entry.eigo.length ? entry.eigo : '-') + '\n');
+      result.push(entry.kanji + '\n');
+      result.push((entry.eigo.length ? entry.eigo : '-') + '\n');
 
-      b.push(entry.onkun.replace(/\.([^\u3001]+)/g, '\uFF08$1\uFF09') + '\n');
+      result.push(
+        entry.onkun.replace(/\.([^\u3001]+)/g, '\uFF08$1\uFF09') + '\n'
+      );
       if (entry.nanori.length) {
-        b.push('\u540D\u4E57\u308A\t' + entry.nanori + '\n');
+        result.push('\u540D\u4E57\u308A\t' + entry.nanori + '\n');
       }
       if (entry.bushumei.length) {
-        b.push('\u90E8\u9996\u540D\t' + entry.bushumei + '\n');
+        result.push('\u90E8\u9996\u540D\t' + entry.bushumei + '\n');
       }
 
-      for (i = 0; i < this.kanjiInfoLabelList.length; i += 2) {
-        e = this.kanjiInfoLabelList[i];
-        j = entry.misc[e];
-        b.push(
-          this.kanjiInfoLabelList[i + 1].replace('&amp;', '&') +
-            '\t' +
-            (j || '-') +
-            '\n'
+      for (let i = 0; i < this.kanjiInfoLabelList.length; i += 2) {
+        const kanjiInfoCode = this.kanjiInfoLabelList[i];
+        const kanjiInfoName = this.kanjiInfoLabelList[i + 1];
+        const kanjiInfo = entry.misc[kanjiInfoCode];
+        result.push(
+          kanjiInfoName.replace('&amp;', '&') + '\t' + (kanjiInfo || '-') + '\n'
```

Comment:
Thinking about these hyphens. I wonder if it would be better to replace empty string with hyphens from the very beginning (when we read the data from the dictionary) so we don't need to check so often later.

This one could at least be moved to line 966 but that depends on how things look after later refactors.

Reply:
This is a good idea. I decided to do this with both `kanjiInfo` and `kanjiInfoName`.